apollo-http-cache/src/main/kotlin/com/apollographql/apollo3/cache/http/CachingHttpInterceptor.kt

```diff
@@ -48,17 +49,31 @@ class CachingHttpInterceptor(
         return networkMightThrow(request, chain, cacheKey)
       }
       NETWORK_FIRST -> {
+        var networkException: ApolloException? = null
         try {
           val response = networkMightThrow(request, chain, cacheKey)
           if (response.statusCode in 200..299) {
             //  let HTTP errors through
             return response
           }
```

Comment:
You can throw an ApolloHttpResponse here:

```suggestion
          if (response.statusCode in 200..299) {
            //  let HTTP errors through
            return response
          } else {
            throw ApolloHttpException(response.statusCode, response.headers)
          }
```

That'll simplify the block below